src/state/achievements/helpers.ts

```diff
@@ -1,25 +1,38 @@
-import { getProfileContract } from 'utils/contractHelpers'
+import { request, gql } from 'graphql-request'
 import { campaignMap } from 'config/constants/campaigns'
 import { Achievement } from 'state/types'
 import { getAchievementTitle, getAchievementDescription } from 'utils/achievements'
 
+interface UserPointIncreaseEvent {
+  campaignId: string
+  id: string // wallet address
+  points: string
+}
+
+const profileSubgraphApi = process.env.REACT_APP_SUBGRAPH_PROFILE
+
 /**
  * Gets all user point increase events on the profile filtered by wallet address
  */
-export const getUserPointIncreaseEvents = async (account: string) => {
+export const getUserPointIncreaseEvents = async (account: string): Promise<UserPointIncreaseEvent[]> => {
   try {
-    const profileContract = getProfileContract()
-    const events = await profileContract.getPastEvents('UserPointIncrease', {
-      fromBlock: 'earliest',
-      toBlock: 'latest',
-      filter: {
-        userAddress: account,
-      },
-    })
-
-    return events
+    const data = await request(
+      profileSubgraphApi,
+      gql`
+        {
+          user(id: "${account}") {
+            points {
+              id
+              campaignId
+              points
+            }
+          }
+        }
+      `,
+    )
+    return data.user.points
```

Comment:
Deliberately kept it simple so we can test and adapt our graphql architectural strategy.